### DISCUSSION TITLE
Assign different properties (material constants)

### DISCUSSION THREAD

**Discussion by fabiomachadofonseca** [thelfer/tfel#368] [Q&A] [answered]
Hello,

Before all, thanks for creating this enviroment to democratize the behavior's law development!

I am currently using MFront coupled with Abaqus to model a trilayer behavior with a modified chaboche's law. I would like to know if it is possible to assign different properties (with the same law) for each layer. For example, layer one E = 150000, layer two E = 180000 and layer three E = 210000.

Attached you'll see my .inp example for one layer.

Best,
Fabio
[Example.txt](https://github.com/thelfer/tfel/files/10869805/Example.txt)

**Comment by thelfer** [comment, marked as answer]
@fabiomachadofonseca By layers,  you mean several materials ? If so, you may have a look at the documentation (https://thelfer.github.io/tfel/web/abaqus.html):

> The function name includes the modelling hypothesis, see below. An identifier can optionnaly be added to reuse the same
 > behaviour for several material (with different material properties for instance). The identifier is discarded in
> the umat.cpp, vumat-sp.cpp and and vumat-dp.cpp files.
>
> Thus, the material name in Abaqus/Standard and Abaqus/Explicit is expected to have the following form:
> LIBRARY_FUNCTION_HYPOTHESIS_IDENTIFIER.

In your case, you can do something like:

~~~~
*Solid Section, elset=_EL1, material=ABAQUSBEHAVIOUR_CHABOCHE_FABIO_3D_LAYER1
*Solid Section, elset=_EL2, material=ABAQUSBEHAVIOUR_CHABOCHE_FABIO_3D_LAYER2
....
~~~~

and declare the appropriate `*Material` sections

**Reply by fabiomachadofonseca** [reply]
@thelfer yes! I mean materials, I was with this image on my head:
![image](https://user-images.githubusercontent.com/126767246/222425049-60ce75b2-8d07-4df6-9f88-27c26fe4b201.png)

Thank you for your help, it works perfectly :)

Best,
Fabio

**Reply by thelfer** [reply]
Perfect ! To be honest we did the `Abaqus` interface quite a long time ago, so I myself forgot about this feature. Let us know if you make progresses. Do not hesitate to share your works at the MFront User days.
Best,
Thomas

**Reply by fabiomachadofonseca** [reply]
This feature will be also useful to work with crystal plasticity! I will let you know about the progresses ;)

A+
Fabio
